tests/testthat/_snaps/property.md

```diff
@@ -17,3 +17,26 @@
     Error <simpleError>
       trying to get slot "blah" from an object of a basic class ("NULL") with no slots
 
+# properties can be base, S3, S4, R7, or R7 union
+
+    Code
+      my_obj@base <- "x"
+    Error <simpleError>
+      <my_class>@base must be of class <integer>, not <character>
+    Code
+      my_obj@s3 <- "x"
+    Error <simpleError>
+      <my_class>@s3 must be of class <factor>, not <character>
+    Code
+      my_obj@s4 <- "x"
+    Error <simpleError>
+      <my_class>@s4 must be of class <class_s4>, not <character>
+    Code
+      my_obj@r7 <- "x"
+    Error <simpleError>
+      <my_class>@r7 must be of class <class_r7>, not <character>
+    Code
+      my_obj@r7_union <- "x"
+    Error <simpleError>
+      <my_class>@r7_union must be of class <integer> u <logical>, not <character>
```

Comment:
How do you feel about a capital U? `<integer> U <logical>`. I don't really mind either way but I think it looks slightly more like the typical union symbol

Reply:
I was actually thinking about switching to `|` . What do you think about that?

Reply:
I also don't mind `<union<logical, integer>>`, which makes sense if you consider that union is just another "type" so it should get wrapped in `< >`.

I think something that is currently missing from the current approach with either `U` or `|` is some kind of signal for the start / end of the union - which would be the `<union` and then the final `>` in the other approach I suggested.

I don't mind the `|` that much though, if you want to go with that

Reply:
I was thinking it doesn't need a signal here because unions can't be nested (they're always automatically flattened), so it's just enumerating the set of possible values here.

Maybe make it `<my_class>@r7_union must be of class <integer> or <logical>, not <character>`

Reply:
Okay, I think enumerating is fine, I like the `or` approach. Possibly with support for >=3 like `<character>, <double> or <integer>` depending on your Oxford comma preference